core/src/main/java/com/rudderstack/android/sdk/core/RudderDeviceModeManager.java

```diff
@@ -277,11 +298,39 @@ void dumpTransformedEvents(TransformationResponse transformationResponse) {
                 return;
             List<TransformedEvent> transformedEvents = transformedDestination.payload;
             sortTransformedEventBasedOnOrderNo(transformedEvents);
-            for (TransformedEvent transformedEvent : transformedEvents) {
-                if (transformedEvent.status.equals("200")) {
-                    dumpEventToDestinations(transformedEvent.event, Collections.singletonList(destinationName), "dumpTransformedEvents");
+            sendEventsToTransformedDestinations(transformedDestination, destinationName);
+        }
+    }
+
+    private void sendEventsToTransformedDestinations(TransformedDestination transformedDestination, String destinationName) {
+        for (TransformedEvent transformedEvent : transformedDestination.payload) {
```

Comment:
Are we sure `transformedDestination.payload` is never null?

Reply:
I'm going to implement a check for nullability of `transformedDestination.payload`.